x-pack/plugins/fleet/server/services/artifacts/artifacts.ts

```diff
@@ -281,3 +284,102 @@ export const encodeArtifactContent = async (
 
   return encodedArtifact;
 };
+
+/**
+ * Returns an iterator that loops through all the artifacts stored in the index
+ *
+ * @param esClient
+ * @param options
+ *
+ * @example
+ *
+ * async () => {
+ *   for await (const value of fetchAllArtifactsIterator()) {
+ *     // process page of data here
+ *   }
+ * }
+ */
+export const fetchAllArtifacts = (
```

Comment:
Fleet team: FYI - the PR for `main` is here: https://github.com/elastic/kibana/pull/174411

The implementation differs slightly in that I instead created generic/reusable utilities for both ES and SO clients that return back an `AsyncIterator`.  But the logic is the same

Reply:
Is there any reason why you are not backporting the same pr?

Reply:
The PR for `main` will be more involved as we will look into other areas of our `ManifestManger` to ensure we can process large amounts of data more efficiently. We wanted to keep this change for 8.12 small and focused on the issue that was reported by one of our customers.